Userland/Libraries/LibJS/AST.cpp

```diff
@@ -2446,7 +2440,32 @@ Value SwitchStatement::execute(Interpreter& interpreter, GlobalObject& global_ob
                 }
             }
         }
+        return {};
+    };
+
+    bool falling_through = false;
+    SwitchCase const* default_switch_case = nullptr;
+    for (auto& switch_case : m_cases) {
+        if (!switch_case.test()) {
+            default_switch_case = &switch_case;
+            continue;
+        }
+        if (!falling_through) {
+            auto test_result = switch_case.test()->execute(interpreter, global_object);
+            if (interpreter.exception())
+                return {};
+            if (!is_strictly_equal(discriminant_result, test_result))
+                continue;
+        }
+        falling_through = true;
+        if (auto result = execute_switch_case(switch_case); result.has_value())
+            return *result;
```

Comment:
Wouldn't we end up trying to execute more cases if this throws?

Reply:
nah, it returns an empty JS::Value in that case

Reply:
Ah right, my bad :P